arrow-row/src/lib.rs

```diff
@@ -1002,6 +1002,46 @@ impl Rows {
             + self.buffer.len()
             + self.offsets.len() * std::mem::size_of::<usize>()
     }
+
+    /// Return a new Rows sliced according to `offset` and `length`
+    /// This copies memory from the original Rows
+    ///
+    /// # Panics
+    ///
+    /// Panics if `offset` with `length` is greater than row length.
+    pub fn slice(&self, offset: usize, length: usize) -> Self {
+        assert!(
+            offset + length <= self.num_rows(),
```

Comment:
```suggestion
            offset.saturating_add(length) <= self.num_rows(),
```
Otherwise this will misbehave on overflow